### DISCUSSION TITLE
Crawler configs and legacy docsearch v2 + v3 opening in dropdown possible?

### DISCUSSION THREAD

**Discussion by remcokalf** [algolia/docsearch#1878] [Q&A] [answered]
Not sure if this is the right place to ask. If not, please let me know where.

We have 2 implementations of Docsearch on our site: one is v3, which has its configs set in crawler.algolia.com. 

The other one is still on v2. That one's configs are set in a json file in this archived repo: https://github.com/algolia/docsearch-configs/blob/master/configs/
There is no related config in crawler.algolia.com.

We'd like to update the v2 one to v3, but we can't because the v3 implementation opens in a modal, which does not work in the iframe that this app section is running in. So we are stuck with v2. On top of that we cannot update the configs for this search.

So my question is three-fold:

1/ Is it in any way possible to have v3 docsearch open as a dropdown, like v2 does? If possible, we could simply update to v3.

2/ If we (have to) stick with v2, is it possible to let it use the config settings in crawler.algolia.com, or somewhere else, instead of the json file in the archived legacy repo?

3/ If the previous does does not work, we'll have to remain at v2. How can we change the config for the v2 version, as that is an archived repo?

Thanks!

**Comment by remcokalf** [comment]
Are these discussions even read? Should I post this in issues?

**Comment by shortcuts** [comment]
Hey, sorry for the delay, I did not saw it

> 1/ Is it in any way possible to have v3 docsearch open as a dropdown, like v2 does? If possible, we could simply update to v3.

No, it can't (cc @shaneafsar @8bittitan in case you want to add something)

> 2/ If we (have to) stick with v2, is it possible to let it use the config settings in crawler.algolia.com, or somewhere else, instead of the json file in the archived legacy repo?

The records outputted from the Crawler should work in most cases with DocSearch v2, it might be worth applying and then porting the settings there so that you can use the new infra but keep v2 if you want 

> 3/ If the previous does does not work, we'll have to remain at v2. How can we change the config for the v2 version, as that is an archived repo?

It's possible to make the Crawler work with DocSearch v2

**Comment by remcokalf** [comment]
Hi, thanks for your reply.

Good news that it should be possible to use the crawler records with v2. 

However, I already tried using v3 settings but could not get it to work.

This is our v3 implementation:
```
<script>
    (function ($) {
        $(function () {
            docsearch({
                container: '#docsearch',
                appId: 'xxx',
                apiKey: 'xxx',
                indexName: 'xxx',
                placeholder: 'Search the Help Center ...',
                translations: {
                    button: {
                        buttonText: 'Click to search ...'
                    }
                },
                getMissingResultsUrl({ query }) {
                    return `mailto:support@xx.com?subject=Feedback: Help Center Search does not retrieve results for: ${query}`;
                },
                searchParameters: {
                    hitsPerPage: 30
                }
            });
        });
    })(fUtil);
</script>
```

This is our v2 implementation:
```
<script type="text/javascript">
var search = docsearch({
    apiKey: 'xxx',
    indexName: 'xxx',
    inputSelector: '#Keywords',
    debug: false,
    handleSelected: function(input, event, suggestion) {
        window.open(suggestion.url, '_blank');
    }
});
</script>
```

Could you give an example of how to use the v3 crawler output in the v2 implementation, based on the above?

If this would work I'm also wondering about the crawler settings. I have set up several levels:
https://d.pr/i/ax5hmT
It would be great if the v2 dropdown would show them.

**Comment by randombeeper** [comment, marked as answer]
Hi @remcokalf obviously quite a long time has passed and I'm new at Algolia so seeing this for the first time. If you still have any issues, I recommend finding me at jamesatalgolia on our [Discord](https://alg.li/discord).

**Reply by remcokalf** [reply]
Hi @randombeeper / James, thanks for picking this up! The issue still exists, but I more or less gave up on it because I got no reaction. However, as it stands now we decided last week that we will probably move our app implementation from iframe to Ajax. As soon as that happens we can move to the v3 modal without issues. If not, I'll message you on Discord. Thanks!

**Comment by randombeeper** [comment]
I'm going to close this discussion as we are trying to keep interactions to our Discord at https://alg.li/discord and we'll be happy to help you there! This discussion area is not well monitored.
